test/unit/logic/broadcaster.js

```diff
@@ -0,0 +1,242 @@
+const rewire = require('rewire');
+
+const Broadcaster = rewire('../../../logic/broadcaster');
+
+describe('Broadcaster', () => {
+	const force = true;
+	const params = { limit: 10, broadhash: '123' };
+	const options = { data: { peer: {}, block: {} }, api: 'blocks' };
+	let broadcaster;
+	let broadcasts;
+	let transactionStub;
+	let peersStub;
+	let loggerStub;
+	let modulesStub;
+	let peerList;
+
+	before(done => {
+		broadcasts = {
+			broadcastInterval: 10000,
+			releaseLimit: 10,
+		};
+
+		peersStub = {
+			me: sinonSandbox.stub().returns(['192.168.10.10']),
+		};
+
+		transactionStub = {
+			checkConfirmed: sinonSandbox
+				.stub()
+				.callsArgWith(1)
+				.returns(null, true),
+		};
+
+		loggerStub = {
+			info: sinonSandbox.stub(),
+			error: sinonSandbox.stub(),
+		};
+
+		peerList = [
+			{
+				rpc: {
+					blocks: sinonSandbox
+						.stub()
+						.callsArgWith(1, new Error('err'))
+						.returns(),
+				},
+			},
+		];
+
+		modulesStub = {
+			peers: {
+				list: sinonSandbox.stub().callsArgWith(1, null, peerList),
+				getLastConsensus: sinonSandbox.stub(),
+			},
+			transport: {},
+			transactions: {
+				transactionInPool: sinonSandbox.stub().returns(false),
+			},
+		};
+
+		broadcaster = new Broadcaster(
+			broadcasts,
+			force,
+			peersStub,
+			transactionStub,
+			loggerStub
+		);
+		broadcaster.bind(
+			modulesStub.peers,
+			modulesStub.transport,
+			modulesStub.transactions
+		);
+		done();
+	});
+
+	after(() => {
+		return sinonSandbox.restore();
+	});
+
+	describe('constructor', () => {
+		it('should return Broadcaster instance', () => {
+			expect(broadcaster).to.be.instanceOf(Broadcaster);
+			expect(broadcaster)
+				.to.have.property('queue')
+				.that.is.an('Array');
+			expect(broadcaster)
+				.to.have.property('config')
+				.that.is.an('object');
+			return expect(broadcaster)
+				.to.have.property('routes')
+				.that.is.an('Array');
+		});
+	});
+
+	describe('enqueue', () => {
+		it('should throw error for no params', () => {
+			return expect(() => {
+				broadcaster.enqueue();
+			})
+				.to.throw()
+				.to.be.instanceOf(Error);
+		});
+
+		it('should push params and options to queue', () => {
+			const params = {};
+			const options = {};
+			expect(broadcaster.enqueue(params, options)).to.eql(1);
+			return expect(broadcaster.enqueue(params, options)).to.eql(2);
+		});
+	});
+
+	describe('getPeers', () => {
+		it('should return peers', done => {
+			broadcaster.getPeers(params, (err, peers) => {
+				expect(err).to.be.null;
+				expect(peers).to.be.an('Array').that.is.not.empty;
+				done();
+			});
+		});
+	});
+
+	describe('broadcast', () => {
+		it('should be able to broadcast block to peers', done => {
```

Comment:
Checking that peer rpc methods are called, blocks are serialized etc should be included in the tests. And there is a bug in broadcast function on line: https://github.com/LiskHQ/lisk/blob/development/logic/broadcaster.js#L194. Here we call the parent `cb` as opposed to `waterCb`, which might lead to memory leaks.